### PR TITLE
Add tests cases for Backtick "`" key

### PR DESCRIPTION
**Pull request by zonble**
The PR adds test cases. Also found a bug that the state did not go back to empty when a user wants to cancel the puncyuation list.

---

This pull request primarily adds comprehensive tests for handling the backtick (`) key and its combinations in both Bopomofo and Plain Bopomofo input modes. It also makes minor formatting and style improvements in the test files, and a small formatting fix in the Objective-C implementation. The new tests ensure that the input method correctly transitions between states when the backtick key is used alone or in combination with other keys like ESC, Backspace, or punctuation.

The most important changes are:

### New tests for backtick (`) key handling

* Added multiple test cases to `KeyHandlerBopomofoTests.swift` to verify correct state transitions when pressing the backtick key in various scenarios, including with buffer, after ESC, after punctuation, and after backspace. These tests cover edge cases to ensure robust handling of the backtick key and state management.
* Added similar backtick key handling tests to `KeyHandlerPlainBopomofoTests.swift`, ensuring consistent behavior across both input modes.

### Test style and formatting improvements

* Updated error callback blocks in `KeyHandlerBopomofoTests.swift` to use explicit braces for clarity and consistency. [[1]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113L2592-R2593) [[2]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113L2612-R2614) [[3]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113L2628-R2631) [[4]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113L2638-R2644) [[5]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113L2648-R2654) [[6]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113L2664-R2671) [[7]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113L2683-R2691) [[8]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113L2702-R2711) [[9]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113L2802-R2812) [[10]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113L2818-R2829) [[11]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113L2831-R2843) [[12]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113L2845-R2858)

### Minor code formatting

* Fixed spacing and formatting in both test files for improved readability. [[1]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113R35) [[2]](diffhunk://#diff-95ade560d07799dcb4604951631304e5a7cf8545524f7fdeff889ba11d3637bfR29) [[3]](diffhunk://#diff-e6bfaa3af46329addae5f6b856521a950a27a4481472d20e628ed96ea0957113L2401)

### Objective-C implementation formatting

* Made a minor formatting improvement in the Objective-C method `_handleCandidateState:` by aligning the callback block for better readability.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "McBopomofo",
+  "lockfileVersion": 3,
+  "requires": true,
+  "packages": {}
+}
```
Comment (ChiahongHong) [inline package-lock.json]: Hi @zonble was this `package-lock.json` committed by accident? McBopomofo shouldn’t have any npm dependencies XD

**Review by lukhnos — APPROVED** [review]
LGTM.
